src/client/Web3SwapClient.ts

```diff
@@ -51,7 +52,32 @@ export class Web3SwapClient {
         }
     }
 
-    private async gatherEvents(provider: providers.Provider, startBlock: number | undefined, contract: Contract, chain: number) {
+    public async isValidOrder(fullOrder: DbOrder) {
+        let isValid = true;
+        const contract = this.contracts[fullOrder.chainId];
+        if (!contract) {
+            return Promise.resolve(false);
+        }
+        try {
+            const response = await contract.check(
+                fullOrder.sender.wallet,
+                fullOrder
+            )
+            const errors = checkResultToErrors(response[1], response[0])
+            for (let index in errors) {
+                if(Object.keys(OrderErrors).includes(errors[index])) {
+                    isValid = false;
+                    break;
+                }                
+            }
```

Comment:
```suggestion
isValid = !errors.some(error => Object.keys(OrderErrors).includes(error));
```

Reply:
`some` is an array function that returns true if condition is met, then it breaks, some behaviour as your for loop.

Reply:
Ho thank you this is much better :D !